syntax = "proto3";

package product_favorite;

service FavoriteProduct {
    rpc CreateFavoriteProduct(CreateFavoriteProductRequest) returns (CreateFavoriteProductResponse) {}
    rpc ListFavoriteProduct(ListFavoriteProductRequest) returns (ListFavoriteProductResponse) {}
}

message CreateFavoriteProductRequest {
    int32 client_id = 1;
    int32 product_id = 2;
}

message CreateFavoriteProductResponse {
    int32 favorite_product_id = 1;
}

message ListFavoriteProductRequest {
    int32 page = 1;
    int32 limit = 2;
    int32 client_id = 3;
}

message FavoriteProductObject {
    int32 id = 1;
    string title = 2;
    float price = 3;
    string image = 5;
}

message ListFavoriteProductResponse {
    FavoriteProductObject products = 1;
}